src/main/java/com/exasol/adapter/dynamodb/queryrunner/DynamodbGetItemQueryPlanFactory.java

```diff
@@ -0,0 +1,114 @@
+package com.exasol.adapter.dynamodb.queryrunner;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.exasol.adapter.dynamodb.documentnode.dynamodb.DynamodbNodeToAttributeValueConverter;
+import com.exasol.adapter.dynamodb.documentnode.dynamodb.DynamodbNodeVisitor;
+import com.exasol.adapter.dynamodb.documentpath.DocumentPathExpression;
+import com.exasol.adapter.dynamodb.dynamodbmetadata.DynamodbTableMetadata;
+import com.exasol.adapter.dynamodb.remotetablequery.*;
+
+/**
+ * This class builds a {@link DynamodbGetItemQueryPlan} if possible.
+ */
+public class DynamodbGetItemQueryPlanFactory {
+
+    /**
+     * Builds a {@link DynamodbGetItemQueryPlan} is possible for the given query.
+     * 
+     * @param documentQuery query to build the plan for
+     * @param tableMetadata DynamoDB table metadata used for checking the primary key
+     * @return the generated plan
+     * @throws PlanDoesNotFitException if this query can't be executed using a {@link DynamodbGetItemQueryPlan}
+     */
+    public DynamodbGetItemQueryPlan buildGetItemPlanIfPossible(
+            final RemoteTableQuery<DynamodbNodeVisitor> documentQuery, final DynamodbTableMetadata tableMetadata)
+            throws PlanDoesNotFitException {
+        final Visitor visitor = new Visitor(tableMetadata);
+        try {
+            documentQuery.getSelection().accept(visitor);
+            final Map<String, AttributeValue> key = visitor.getPrimaryKey();
+            checkIfKeyIsComplete(key, tableMetadata);
```

Comment:
If `documentQuery.getSelection().accept(visitor);` does not throw, move the try before `checkIfKeyIsComplete(..)` call.
This reduces try-catch scope.